cnspec crashes if private key is malformed

**Describe the bug**
If a user fat fingers the private key and it can't be parsed as a valid private key then cnspec crashes:

```
 !  ~/dev  cnspec status
→ loaded configuration from /Users/tsmith/.config/mondoo/mondoo.yml using source default
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x10522018c]

goroutine 1 [running]:
go.mondoo.com/ranger-rpc.(*Client).DoClientRequest(0x14001172140, {0x10d3f5bb0?, 0x14000198000}, {0x10d359280, 0x14001738570}, {0x14000d7c0c0, 0x2f}, {0x10d354b60, 0x140017385a0}, {0x10d354b80, ...})
	/Users/tsmith/go/pkg/mod/go.mondoo.com/ranger-rpc@v0.5.1-0.20220923135836-9e7732899d34/client.go:65 +0x2ac
go.mondoo.com/cnquery/upstream.(*AgentManagerClient).PingPong(0x14001172140, {0x10d3f5bb0, 0x14000198000}, 0x14001738570?)
	/Users/tsmith/dev/cnquery/upstream/upstream.ranger.go:65 +0xc0
go.mondoo.com/cnspec/apps/cnspec/cmd.glob..func26(0x11106c920?, {0x10a9763dd?, 0x0?, 0x0?})
	/Users/tsmith/dev/cnspec/apps/cnspec/cmd/status.go:99 +0x58c
github.com/spf13/cobra.(*Command).execute(0x11106c920, {0x111265cf0, 0x0, 0x0})
	/Users/tsmith/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x5b0
github.com/spf13/cobra.(*Command).ExecuteC(0x11106c360)
	/Users/tsmith/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x35c
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/tsmith/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
go.mondoo.com/cnspec/apps/cnspec/cmd.Execute()
	/Users/tsmith/dev/cnspec/apps/cnspec/cmd/root.go:76 +0x28
main.main()
	/Users/tsmith/dev/cnspec/apps/cnspec/cnspec.go:6 +0x1c
```

**To Reproduce**
Steps to reproduce the behavior:
1. Register cnspec
2. Run `cnspec status` and everything is fine
3. Edit the private key in the config and add some extra characters
4. Run `cnspec status` and what it crash

Similar segv found when testing a new [cnspec policy](https://github.com/mondoohq/cnspec-policies/pull/108), diffrent part of code but likely correlated.  Found on cnspec 7.12.1:

```
$ cnspec scan github repo mondoohq/installer -f Downloads/mondoo-github-security.mql.yaml --incognito
→ no Mondoo configuration file provided. using defaults
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
→ connecting to asset mondoohq/installer (api)

█████████████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░  29% mondoohq/installer
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x686751e]

goroutine 139 [running]:
go.mondoo.com/cnquery/resources/packs/github.(*mqlGithubBranch).GetProtectionRules(0xc0007300f0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/packs/github/github.go:1060 +0x21e
go.mondoo.com/cnquery/resources/packs/github.(*mqlGithubBranch).ComputeProtectionRules(0xc0007300f0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/packs/github/github.lr.go:6300 +0x55
go.mondoo.com/cnquery/resources/packs/github.(*mqlGithubBranch).MqlCompute(0xc00136e340?, {0xc000d06150, 0xf})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/packs/github/github.lr.go:6280 +0x1d4
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc00136e380, {0x9991220, 0xc0007300f0}, {0xc000d06150, 0xf}, {0xc001b2eb00, 0x32}, 0xc001bec600)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:345 +0x39b
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000a1f730, 0xc001bec390, 0xc001457db0, 0x300000002)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000a1f730, 0xc001bec390, 0xc001457db0, 0xc001bccb40?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000a1f730, 0xc001457db0, 0x30?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000a1f730, 0xc001bec4b0?, 0x300000002?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001bea660?, 0xc?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000a1f730, 0x8808ee6?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000a1f730)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlock(0xc000a2b5e0, {0xc001be21a0, 0x1, 0xc00065b280?}, 0x300000000, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:582 +0x12f
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks(0xc000a2b5e0, {0xc0006bdda0, 0x1, 0x0?}, 0x5?, 0xc0013cc3c0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:553 +0x11b
go.mondoo.com/cnquery/llx._arrayWhereV2(0xc000a2b5e0, 0xc001bec2d0, 0xc0014f2050, 0x100000004, 0x1)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:275 +0x4d7
go.mondoo.com/cnquery/llx.arrayWhereNotV2(0xc000d1d9e0?, 0xf?, 0x880f75d?, 0x9?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:307 +0x1e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000a2b5e0, 0xc001bec2d0, 0xc0014f2050, 0xc0013cc594?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:782 +0x119
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000a2b5e0, 0xc0014f2050, 0x5a1ed45?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000a2b5e0, 0xc001a6aab0?, 0x100000004?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001a65570?, 0x5a10dce?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000a2b5e0, 0xc001a6aa80?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000a2b5e0, 0x1?, 0xc001bec2d0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx._arrayWhereV2.func1({0xc001be0000?, 0x13?, 0xc001459a40?}, {0x58?, 0xc000654c80?, 0xd6f5440?})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:292 +0x19f
go.mondoo.com/cnquery/llx.(*arrayBlockCallResults).update(0xc000a2b650, 0x12, 0xc000654c80)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:488 +0x41a
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks.func1(0xc000654c80?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:554 +0x27
go.mondoo.com/cnquery/llx.reportSync.func1(0x794bae0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:370 +0xa2
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000a1e9a0, 0xc001be5fb0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:852 +0x39f
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000a1e9a0, 0xc001be5fb0?, 0xc001bec210)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx.runResourceFunction.func1({0x759a180?, 0xd474720}, {0x0?, 0x0})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:731 +0x245
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc00136e380, {0x9991220, 0xc0001a2248}, {0xc000d060f0, 0x9}, {0xc001b2ea80, 0x31}, 0xc001bec150)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:360 +0x449
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000a1e9a0, 0xc001bde1e0, 0xc001457bd0, 0x200000002)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000a1e9a0, 0xc001bde1e0, 0xc001457bd0, 0xc0013ccea8?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000a1e9a0, 0xc001457bd0, 0x30?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000a1e9a0, 0xc001bec000?, 0x200000002?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001bea4c0?, 0xa?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000a1e9a0, 0x8808ee6?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000a1e9a0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlock(0xc000a2b5e0, {0xc000575ef8, 0x1, 0xc00065ad40?}, 0x200000000, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:582 +0x12f
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks(0xc000a2b5e0, {0xc00165e780, 0x13, 0x0?}, 0x0?, 0xc0013cd218?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:553 +0x11b
go.mondoo.com/cnquery/llx._arrayWhereV2(0xc000a2b5e0, 0xc001bc7da0, 0xc001457c70, 0x100000003, 0x0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:275 +0x4d7
go.mondoo.com/cnquery/llx.arrayWhereV2(0xc000d1d9e0?, 0xf?, 0xc000d060b8?, 0x5?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:303 +0x1b
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000a2b5e0, 0xc001bc7da0, 0xc001457c70, 0x40d6bf?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:782 +0x119
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000a2b5e0, 0xc001457c70, 0xd6f5440?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000a2b5e0, 0xc001a6aab0?, 0x100000003?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001a65570?, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000a2b5e0, 0xc001a6aa80?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000a2b5e0, 0xc001a6aa80?, 0xc001bc7da0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx.runResourceFunction.func1({0x75318e0?, 0xc000729e48}, {0x0?, 0x0})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:731 +0x245
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc00136e380, {0x9991360, 0xc000aca0f0}, {0xc000d060b0, 0x8}, {0xc001271d80, 0x31}, 0xc001bc7ce0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:360 +0x449
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000a2b5e0, 0xc001bc7c50, 0xc001457a90, 0x100000002)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000a2b5e0, 0xc001bc7c50, 0xc001457a90, 0xa?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000a2b5e0, 0xc001457a90, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000a2b5e0, 0xc001a6aab0?, 0x100000002?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001a65570?, 0xc001a6aa80?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000a2b5e0, 0x8808ee6?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000a2b5e0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*MQLExecutorV2).Run(0xc0012fb9e0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:283 +0x59
go.mondoo.com/cnspec/policy/executor/internal.(*executionManager).executeCodeBundle(0xc0016ac050, 0xc0013ae960, 0x0?, {0x0, 0x0})
        /home/benr/actions-runner/_work/cnspec/cnspec/policy/executor/internal/execution_manager.go:164 +0x357
go.mondoo.com/cnspec/policy/executor/internal.(*executionManager).Start.func1()
        /home/benr/actions-runner/_work/cnspec/cnspec/policy/executor/internal/execution_manager.go:85 +0x1cb
created by go.mondoo.com/cnspec/policy/executor/internal.(*executionManager).Start
        /home/benr/actions-runner/_work/cnspec/cnspec/policy/executor/internal/execution_manager.go:54 +0x6a
```

@benr I also ran into this ...

```
cnspec scan github repo lunalectric/compliant-repository -f github.mql.yaml --incognito
→ no Mondoo configuration file provided. using defaults
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=0
x could not resolve asset error="a valid GitHub token is required, pass --token '<yourtoken>' or set GITHUB_TOKEN environment variable" asset=
FTL failed to run scan error="failed to resolve multiple assets"
scottford@cloudshell:~ (mondoo-demo-320917)$ export GITHUB_TOKEN=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
scottford@cloudshell:~ (mondoo-demo-320917)$ cnspec scan github repo lunalectric/compliant-repository -f github.mql.yaml --incognito
→ no Mondoo configuration file provided. using defaults
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
→ connecting to asset Lunalectric/compliant-repository (api)

██████████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░  19% Lunalectric/compliant-repository
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x686751e]

goroutine 65 [running]:
go.mondoo.com/cnquery/resources/packs/github.(*mqlGithubBranch).GetProtectionRules(0xc001c5a1e8)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/packs/github/github.go:1060 +0x21e
go.mondoo.com/cnquery/resources/packs/github.(*mqlGithubBranch).ComputeProtectionRules(0xc001c5a1e8)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/packs/github/github.lr.go:6300 +0x55
go.mondoo.com/cnquery/resources/packs/github.(*mqlGithubBranch).MqlCompute(0xc000c8c480?, {0xc001bb0da0, 0xf})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/packs/github/github.lr.go:6280 +0x1d4
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc000c8c4c0, {0x9991220, 0xc001c5a1e8}, {0xc001bb0da0, 0xf}, {0xc0015dc180, 0x32}, 0xc0010b94a0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:345 +0x39b
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000520fc0, 0xc0010b9230, 0xc001c325f0, 0x300000002)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000520fc0, 0xc0010b9230, 0xc001c325f0, 0xc0010ba070?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000520fc0, 0xc001c325f0, 0x30?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000520fc0, 0xc0010b9350?, 0x300000002?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc0010b5020?, 0xc?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000520fc0, 0x8808ee6?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000520fc0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlock(0xc000d61650, {0xc001c5a888, 0x1, 0xc001ceca80?}, 0x300000000, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:582 +0x12f
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks(0xc000d61650, {0xc000874d68, 0x1, 0x0?}, 0x7?, 0xc001db62d0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:553 +0x11b
go.mondoo.com/cnquery/llx._arrayWhereV2(0xc000d61650, 0xc0010b9170, 0xc001c32870, 0x100000004, 0x1)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:275 +0x4d7
go.mondoo.com/cnquery/llx.arrayWhereNotV2(0xc000acd9b0?, 0xf?, 0x880f75d?, 0x9?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:307 +0x1e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000d61650, 0xc0010b9170, 0xc001c32870, 0xc001db64a4?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:782 +0x119
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000d61650, 0xc001c32870, 0x5a1ed45?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000d61650, 0xc000da6030?, 0x100000004?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001971490?, 0x5a10dce?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000d61650, 0xc000da6000?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000d61650, 0x1?, 0xc0010b9170)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx._arrayWhereV2.func1({0xc0010b4e30?, 0x1?, 0xc001c0f080?}, {0x58?, 0xc001cc2c60?, 0x8?})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:292 +0x19f
go.mondoo.com/cnquery/llx.(*arrayBlockCallResults).update(0xc0005202a0, 0x0, 0xc001cc2c60)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:488 +0x41a
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks.func1(0xc001cc2c60?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:554 +0x27
go.mondoo.com/cnquery/llx.reportSync.func1(0x794bae0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:370 +0xa2
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000520d90, 0xc0010b8e10?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:852 +0x39f
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000520d90, 0xc0010b8e10?, 0xc0010b90b0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx.runResourceFunction.func1({0x759a180?, 0xd474728}, {0x0?, 0x0})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:731 +0x245
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate.func1()
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:315 +0x258
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc000c8c4c0, {0x9991220, 0xc001c5a1e8}, {0xc001bb0d80, 0x9}, {0xc0015dc080, 0x31}, 0xc0010b8f90)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:354 +0x3f4
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000520d90, 0xc0010b8d20, 0xc001c32410, 0x200000002)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000520d90, 0xc0010b8d20, 0xc001c32410, 0xc0010ba030?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000520d90, 0xc001c32410, 0x30?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000520d90, 0xc0010b8e40?, 0x200000002?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc0010b4e40?, 0xa?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000520d90, 0x8808ee6?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000520d90)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlock(0xc000d61650, {0xc001c5a858, 0x1, 0xc001ceca00?}, 0x200000000, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:582 +0x12f
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunctionBlocks(0xc000d61650, {0xc000874c00, 0x1, 0x0?}, 0x1?, 0xc001db7218?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:553 +0x11b
go.mondoo.com/cnquery/llx._arrayWhereV2(0xc000d61650, 0xc0010b8c60, 0xc001c324b0, 0x100000003, 0x0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:275 +0x4d7
go.mondoo.com/cnquery/llx.arrayWhereV2(0xc000acd9b0?, 0xf?, 0xc001bb0d78?, 0x5?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin_array.go:303 +0x1b
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000d61650, 0xc0010b8c60, 0xc001c324b0, 0x1f1be40?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:782 +0x119
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000d61650, 0xc001c324b0, 0x41720b?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000d61650, 0xc000da6030?, 0x100000003?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001971490?, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000d61650, 0xc000da6000?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).triggerChain(0xc000d61650, 0xc000da6000?, 0xc0010b8c60)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:910 +0x399
go.mondoo.com/cnquery/llx.runResourceFunction.func1({0x75318e0?, 0xc001ccad80}, {0x0?, 0x0})
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:731 +0x245
go.mondoo.com/cnquery/resources.(*Runtime).WatchAndUpdate(0xc000c8c4c0, {0x9991360, 0xc001c5a480}, {0xc001bb0d70, 0x8}, {0xc0015dc000, 0x31}, 0xc0010b8bd0)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/resources/runtime.go:360 +0x449
go.mondoo.com/cnquery/llx.runResourceFunction(0xc000d61650, 0xc0010b8b40, 0xc001c322d0, 0x100000002)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:712 +0x23e
go.mondoo.com/cnquery/llx.(*blockExecutor).runBoundFunction(0xc000d61650, 0xc0010b8b40, 0xc001c322d0, 0x2?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/builtin.go:795 +0xf2
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc000d61650, 0xc001c322d0, 0x0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:764 +0xde
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc000d61650, 0xc000da6030?, 0x100000002?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc001971490?, 0xc000da6000?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc000d61650, 0x8808ee6?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc000d61650)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*MQLExecutorV2).Run(0xc001bab1a0?)
        /home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20230112143917-2934d0ee356e/llx/llx.go:283 +0x59
go.mondoo.com/cnspec/policy/executor/internal.(*executionManager).executeCodeBundle(0xc001cc5590, 0xc001bc26e0, 0xc000c30ee0?, {0x0, 0x0})
        /home/benr/actions-runner/_work/cnspec/cnspec/policy/executor/internal/execution_manager.go:164 +0x357
go.mondoo.com/cnspec/policy/executor/internal.(*executionManager).Start.func1()
        /home/benr/actions-runner/_work/cnspec/cnspec/policy/executor/internal/execution_manager.go:85 +0x1cb
created by go.mondoo.com/cnspec/policy/executor/internal.(*executionManager).Start
        /home/benr/actions-runner/_work/cnspec/cnspec/policy/executor/internal/execution_manager.go:54 +0x6a
scottford@cloudshell:~ (mondoo-demo-320917)$ 
```

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -136,10 +136,20 @@
         "docker",
         "image",
         "ubuntu:22.04",
-        "--log-level",
-        "debug",
         "--config",
-        "/home/christian/workspace/mondoo/gitlab.com/mondoo/demo.agent.credentials.json"
+        "/tmp/config.yaml"
+      ],
+    },
+    {
+      "name": "Upstream status",
+      "type": "go",
+      "request": "launch",
+      "program": "${workspaceRoot}/apps/cnspec/cnspec.go",
+      "cwd": "${workspaceRoot}/",
+      "args": [
+        "status",
+        "--config",
+        "/tmp/config.yaml"
       ],
     }
   ]

--- a/apps/cnspec/cmd/bundle.go
+++ b/apps/cnspec/cmd/bundle.go
@@ -10,6 +10,7 @@ import (
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	cnquery_cmd "go.mondoo.com/cnquery/apps/cnquery/cmd"
 	cnquery_config "go.mondoo.com/cnquery/apps/cnquery/cmd/config"
 	"go.mondoo.com/cnquery/cli/config"
 	"go.mondoo.com/cnquery/upstream"
@@ -191,7 +192,11 @@ var policyPublishCmd = &cobra.Command{
 			log.Fatal().Msg("cnquery has no credentials. Log in with `cnquery login`")
 		}
 
-		certAuth, _ := upstream.NewServiceAccountRangerPlugin(serviceAccount)
+		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
+		if err != nil {
+			log.Error().Err(err).Msg("could not initialize client authentication")
+			os.Exit(cnquery_cmd.ConfigurationErrorCode)
+		}
 		queryHubServices, err := policy.NewPolicyHubClient(opts.UpstreamApiEndpoint(), ranger.DefaultHttpClient(), certAuth)
 		if err != nil {
 			log.Fatal().Err(err).Msg("could not connect to policy hub")

--- a/apps/cnspec/cmd/bundle.go
+++ b/apps/cnspec/cmd/bundle.go
@@ -194,7 +194,7 @@ var policyPublishCmd = &cobra.Command{
 
 		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
 		if err != nil {
-			log.Error().Err(err).Msg("could not initialize client authentication")
+			log.Error().Err(err).Msg(errorMessageServiceAccount)
 			os.Exit(cnquery_cmd.ConfigurationErrorCode)
 		}
 		queryHubServices, err := policy.NewPolicyHubClient(opts.UpstreamApiEndpoint(), ranger.DefaultHttpClient(), certAuth)

--- a/apps/cnspec/cmd/login.go
+++ b/apps/cnspec/cmd/login.go
@@ -2,6 +2,7 @@ package cmd
 
 import (
 	"context"
+	"os"
 	"strings"
 	"time"
 
@@ -9,6 +10,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"go.mondoo.com/cnquery"
+	cnquery_cmd "go.mondoo.com/cnquery/apps/cnquery/cmd"
 	cnquery_config "go.mondoo.com/cnquery/apps/cnquery/cmd/config"
 	"go.mondoo.com/cnquery/cli/config"
 	"go.mondoo.com/cnquery/cli/sysinfo"
@@ -155,10 +157,10 @@ func register(token string) {
 			plugins = append(plugins, defaultPlugins...)
 			certAuth, err := upstream.NewServiceAccountRangerPlugin(credential)
 			if err != nil {
-				log.Warn().Err(err).Msg("could not initialize certificate authentication")
-			} else {
-				plugins = append(plugins, certAuth)
+				log.Error().Err(err).Msg("could not initialize client authentication")
+				os.Exit(cnquery_cmd.ConfigurationErrorCode)
 			}
+			plugins = append(plugins, certAuth)
 
 			client, err := upstream.NewAgentManagerClient(apiEndpoint, ranger.DefaultHttpClient(), plugins...)
 			if err != nil {
@@ -205,7 +207,8 @@ func register(token string) {
 	plugins = append(plugins, defaultPlugins...)
 	certAuth, err := upstream.NewServiceAccountRangerPlugin(credential)
 	if err != nil {
-		log.Warn().Err(err).Msg("could not initialize certificate authentication")
+		log.Error().Err(err).Msg("could not initialize client authentication")
+		os.Exit(cnquery_cmd.ConfigurationErrorCode)
 	}
 	plugins = append(plugins, certAuth)
 	client, err := upstream.NewAgentManagerClient(apiEndpoint, ranger.DefaultHttpClient(), plugins...)

--- a/apps/cnspec/cmd/login.go
+++ b/apps/cnspec/cmd/login.go
@@ -157,7 +157,7 @@ func register(token string) {
 			plugins = append(plugins, defaultPlugins...)
 			certAuth, err := upstream.NewServiceAccountRangerPlugin(credential)
 			if err != nil {
-				log.Error().Err(err).Msg("could not initialize client authentication")
+				log.Error().Err(err).Msg(errorMessageServiceAccount)
 				os.Exit(cnquery_cmd.ConfigurationErrorCode)
 			}
 			plugins = append(plugins, certAuth)
@@ -207,7 +207,7 @@ func register(token string) {
 	plugins = append(plugins, defaultPlugins...)
 	certAuth, err := upstream.NewServiceAccountRangerPlugin(credential)
 	if err != nil {
-		log.Error().Err(err).Msg("could not initialize client authentication")
+		log.Error().Err(err).Msg(errorMessageServiceAccount)
 		os.Exit(cnquery_cmd.ConfigurationErrorCode)
 	}
 	plugins = append(plugins, certAuth)

--- a/apps/cnspec/cmd/logout.go
+++ b/apps/cnspec/cmd/logout.go
@@ -53,21 +53,21 @@ the credentials cannot be used in future anymore.
 		// check valid client authentication
 		serviceAccount := opts.GetServiceCredential()
 		if serviceAccount == nil {
-			log.Error().Err(err).Msg("could not initialize client authentication")
+			log.Error().Err(err).Msg(errorMessageServiceAccount)
 			os.Exit(cnquery_cmd.ConfigurationErrorCode)
 		}
 
 		plugins := defaultRangerPlugins(sysInfo, opts.GetFeatures())
 		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
 		if err != nil {
-			log.Error().Err(err).Msg("could not initialize client authentication")
+			log.Error().Err(err).Msg(errorMessageServiceAccount)
 			os.Exit(cnquery_cmd.ConfigurationErrorCode)
 		}
 		plugins = append(plugins, certAuth)
 
 		client, err := upstream.NewAgentManagerClient(opts.UpstreamApiEndpoint(), ranger.DefaultHttpClient(), plugins...)
 		if err != nil {
-			log.Error().Err(err).Msg("could not initialize client authentication")
+			log.Error().Err(err).Msg("could not initialize connection to Mondoo Platform")
 			os.Exit(cnquery_cmd.ConfigurationErrorCode)
 		}
 

--- a/apps/cnspec/cmd/root.go
+++ b/apps/cnspec/cmd/root.go
@@ -34,6 +34,10 @@ const cnspecLogo = "  ___ _ __  ___ _ __   ___  ___ \n" +
 	" \\___|_| |_|___/ .__/ \\___|\\___|\n" +
 	"   mondoo™     |_|              "
 
+const (
+	errorMessageServiceAccount = "invalid service account configuration"
+)
+
 func init() {
 	theme.DefaultTheme.Landing = landing()
 	theme.DefaultTheme.Welcome = welcome()

--- a/apps/cnspec/cmd/scan.go
+++ b/apps/cnspec/cmd/scan.go
@@ -12,6 +12,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"go.mondoo.com/cnquery"
+	cnquery_cmd "go.mondoo.com/cnquery/apps/cnquery/cmd"
 	"go.mondoo.com/cnquery/apps/cnquery/cmd/builder"
 	"go.mondoo.com/cnquery/cli/components"
 	"go.mondoo.com/cnquery/cli/config"
@@ -466,7 +467,11 @@ func getCobraScanConfig(cmd *cobra.Command, args []string, provider providers.Pr
 
 	serviceAccount := opts.GetServiceCredential()
 	if serviceAccount != nil {
-		certAuth, _ := upstream.NewServiceAccountRangerPlugin(serviceAccount)
+		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
+		if err != nil {
+			log.Error().Err(err).Msg("could not initialize client authentication")
+			os.Exit(cnquery_cmd.ConfigurationErrorCode)
+		}
 		plugins := []ranger.ClientPlugin{certAuth}
 		// determine information about the client
 		sysInfo, err := sysinfo.GatherSystemInfo()

--- a/apps/cnspec/cmd/scan.go
+++ b/apps/cnspec/cmd/scan.go
@@ -469,7 +469,7 @@ func getCobraScanConfig(cmd *cobra.Command, args []string, provider providers.Pr
 	if serviceAccount != nil {
 		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
 		if err != nil {
-			log.Error().Err(err).Msg("could not initialize client authentication")
+			log.Error().Err(err).Msg(errorMessageServiceAccount)
 			os.Exit(cnquery_cmd.ConfigurationErrorCode)
 		}
 		plugins := []ranger.ClientPlugin{certAuth}

--- a/apps/cnspec/cmd/serve_api.go
+++ b/apps/cnspec/cmd/serve_api.go
@@ -60,7 +60,7 @@ var serveApiCmd = &cobra.Command{
 
 		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
 		if err != nil {
-			log.Error().Err(err).Msg("could not initialize client authentication")
+			log.Error().Err(err).Msg(errorMessageServiceAccount)
 			os.Exit(cnquery_cmd.ConfigurationErrorCode)
 		}
 		plugins := []ranger.ClientPlugin{certAuth}

--- a/apps/cnspec/cmd/serve_api.go
+++ b/apps/cnspec/cmd/serve_api.go
@@ -14,6 +14,7 @@ import (
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	cnquery_cmd "go.mondoo.com/cnquery/apps/cnquery/cmd"
 	"go.mondoo.com/cnquery/cli/config"
 	"go.mondoo.com/cnquery/cli/sysinfo"
 	"go.mondoo.com/cnquery/logger"
@@ -57,7 +58,11 @@ var serveApiCmd = &cobra.Command{
 			log.Fatal().Msg("no service account configured")
 		}
 
-		certAuth, _ := upstream.NewServiceAccountRangerPlugin(serviceAccount)
+		certAuth, err := upstream.NewServiceAccountRangerPlugin(serviceAccount)
+		if err != nil {
+			log.Error().Err(err).Msg("could not initialize client authentication")
+			os.Exit(cnquery_cmd.ConfigurationErrorCode)
+		}
 		plugins := []ranger.ClientPlugin{certAuth}
 		// determine information about the client
 		sysInfo, err := sysinfo.GatherSystemInfo()

--- a/apps/cnspec/cmd/status.go
+++ b/apps/cnspec/cmd/status.go
@@ -11,6 +11,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"go.mondoo.com/cnquery"
+	cnquery_cmd "go.mondoo.com/cnquery/apps/cnquery/cmd"
 	cnquery_config "go.mondoo.com/cnquery/apps/cnquery/cmd/config"
 	"go.mondoo.com/cnquery/cli/config"
 	"go.mondoo.com/cnquery/cli/sysinfo"
@@ -96,7 +97,8 @@ Status sends a ping to Mondoo Platform to verify the credentials.
 
 			certAuth, err := upstream.NewServiceAccountRangerPlugin(credentials)
 			if err != nil {
-				log.Fatal().Err(err).Msg("could not load service account credentials")
+				log.Error().Err(err).Msg("could not initialize client authentication")
+				os.Exit(cnquery_cmd.ConfigurationErrorCode)
 			}
 			plugins = append(plugins, certAuth)
 

--- a/apps/cnspec/cmd/status.go
+++ b/apps/cnspec/cmd/status.go
@@ -94,7 +94,10 @@ Status sends a ping to Mondoo Platform to verify the credentials.
 				s.Client.Mrn = "no managed client"
 			}
 
-			certAuth, _ := upstream.NewServiceAccountRangerPlugin(credentials)
+			certAuth, err := upstream.NewServiceAccountRangerPlugin(credentials)
+			if err != nil {
+				log.Fatal().Err(err).Msg("could not load service account credentials")
+			}
 			plugins = append(plugins, certAuth)
 
 			// try to ping the server

--- a/apps/cnspec/cmd/status.go
+++ b/apps/cnspec/cmd/status.go
@@ -97,7 +97,7 @@ Status sends a ping to Mondoo Platform to verify the credentials.
 
 			certAuth, err := upstream.NewServiceAccountRangerPlugin(credentials)
 			if err != nil {
-				log.Error().Err(err).Msg("could not initialize client authentication")
+				log.Error().Err(err).Msg(errorMessageServiceAccount)
 				os.Exit(cnquery_cmd.ConfigurationErrorCode)
 			}
 			plugins = append(plugins, certAuth)

